exampleconnectionservice/src/main/java/com/twilio/voice/exampleconnectionservice/VoiceConnectionService.java

```diff
@@ -0,0 +1,174 @@
+package com.twilio.voice.exampleconnectionservice;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.content.LocalBroadcastManager;
+import android.telecom.CallAudioState;
+import android.telecom.Connection;
+import android.telecom.ConnectionRequest;
+import android.telecom.ConnectionService;
+import android.telecom.DisconnectCause;
+import android.telecom.PhoneAccountHandle;
+import android.telecom.TelecomManager;
+import android.os.Handler;
+import android.widget.Toast;
+
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.ACTION_DISCONNECT_CALL;
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.ACTION_DTMF_SEND;
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.ACTION_OUTGOING_CALL;
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.CALLEE;
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.DTMF;
+import static com.twilio.voice.exampleconnectionservice.VoiceActivity.OUTGOING_CALL_ADDRESS;
+
+public class VoiceConnectionService extends ConnectionService {
+
+    private static Connection connection;
+
+    public static Connection getConnection() {
+        return connection;
+    }
+
+    public static void deinitConnection() {
+        connection = null;
+    }
+
+    @Override
+    public Connection onCreateIncomingConnection(PhoneAccountHandle connectionManagerPhoneAccount, ConnectionRequest request) {
+        Connection incomingCallCannection = createConnection(request);
+        incomingCallCannection.setRinging();
+        return incomingCallCannection;
+    }
+
+    @Override
+    public Connection onCreateOutgoingConnection(PhoneAccountHandle connectionManagerPhoneAccount, ConnectionRequest request) {
+        Connection outgoingCallConnection = createConnection(request);
+        outgoingCallConnection.setDialing();
+
+        return outgoingCallConnection;
+    }
+
+    private Connection createConnection(ConnectionRequest request) {
+        connection = new Connection() {
+
+            @Override
+            public void onStateChanged(int state) {
+                if (state == Connection.STATE_DIALING) {
+                    final Handler handler = new Handler();
+                    handler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            sendCallRequestToActivity(ACTION_OUTGOING_CALL);
+                        }
+                    });
+                }
+            }
+
+            @Override
+            public void onCallAudioStateChanged(CallAudioState state) {
+                Toast.makeText(getApplicationContext(), "onCallAudioStateChanged called", Toast.LENGTH_SHORT).show();
```

Comment:
Use a regular log message here.

It's not super informative without adding the state to the string as well.